### PR TITLE
Feature/homepage search pokemons

### PR DESCRIPTION
**Pull request by raymond-le-goldenowl**
**Task**
- Type in input then submit to search pokemons, data will display bellow form search

**Descriptions**
- Choose search type (NAME or TYPE) to search pokemons.
- Enter values and submit to search
- After delete all text on input text, data will reset.

**Screenshots**
- Searh type: TYPE
- Input data: bug
Screenshot from 2021-12-30 12-09-23

### CODE REVIEW

**Comment by jun-nguyen-goldenowl** [comment]
please update this PR with branch `main` @raymond-le-goldenowl

File: src/pages/home-page.js [hunk 1]
```diff
@@ -46,41 +53,109 @@ export default function HomePage() {
     setPokemonsDisplay(prePokemons => [...prePokemons, ...pokemonSlice])
   }, [pokemons, offset])
 
+  // Start search when form submit.
+  const handleSubmitSearchForm = e => {
+    e.preventDefault()
+    if (keywords.trim().length > 0) {
+      const stn = 'name'
+      const stt = 'type'
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js:61]: why do we define these constant in here?

File: src/pages/home-page.js [hunk 2]
```diff
@@ -46,41 +53,109 @@ export default function HomePage() {
...
+
+      // check is search by NAME or TYPE
+      if (searchType.type === stn) {
+        const resultFilter = pokemons.filter(pokemon => {
+          return pokemon.name.trim().includes(`${keywords}`.trim())
+        })
+        console.log(resultFilter)
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js]: remove console.log

**Comment by raymond-le-goldenowl** [comment]
I have fix conflict @jun-nguyen-goldenowl

File: src/pages/home-page.js [hunk 3]
```diff
@@ -46,41 +53,109 @@ export default function HomePage() {
...
+        <div className="row">
+          <div className="col-10">
+            <label htmlFor="SearchPokemon" className="form-label">
+              Search
+            </label>
+            <input
+              type="text"
+              className="form-control"
+              id="SearchPokemon"
+              aria-describedby="SearchPokemon"
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js]: the `id` should be in hyphen-separated case

File: src/pages/home-page.js [hunk 4]
```diff
@@ -46,41 +53,109 @@ export default function HomePage() {
...
-        <input
-          type="text"
-          className="form-control"
-          id="search-pokemon-input"
-          aria-describedby="search-pokemon-input"
-        />
+      <form id="form-search" className="mb-5" onSubmit={handleSubmitSearchForm}>
+        <div className="row">
+          <div className="col-10">
+            <label htmlFor="SearchPokemon" className="form-label">
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js]: the `id` should be in hyphen-separated case

File: src/pages/home-page.js [hunk 5]
```diff
@@ -46,41 +53,109 @@ export default function HomePage() {
...
+              type="text"
+              className="form-control"
+              id="SearchPokemon"
+              aria-describedby="SearchPokemon"
+              onChange={e => {
+                setKeywords(e.target.value)
+
+                // reset pokemons while input search form is empty.
+                if (e.target.value.trim() === '') {
+                  console.log(pokemons)
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js]: remove console.log

**Review by jun-nguyen-goldenowl — CHANGES_REQUESTED** [review]

File: src/pages/home-page.js (hunk 1 above)
Comment (raymond-le-goldenowl) [reply src/pages/home-page.js:61]: i think these constant for look clean code.

File: src/pages/home-page.js (hunk 2 above)
Comment (raymond-le-goldenowl) [reply src/pages/home-page.js]: i fixed that

File: src/pages/home-page.js (hunk 4 above)
Comment (raymond-le-goldenowl) [reply src/pages/home-page.js]: i fixed that

File: src/pages/home-page.js (hunk 5 above)
Comment (raymond-le-goldenowl) [reply src/pages/home-page.js]: i fixed that

File: src/pages/home-page.js (hunk 3 above)
Comment (raymond-le-goldenowl) [reply src/pages/home-page.js]: i fixed that

File: src/pages/home-page.js [hunk 6]
```diff
@@ -46,41 +53,107 @@ export default function HomePage() {
...
+              aria-describedby="search-pokemon"
+              onChange={e => {
+                setKeywords(e.target.value)
+
+                // reset pokemons while input search form is empty.
+                if (e.target.value.trim() === '') {
+                  const pokemonSlice = pokemons.slice(0, 0 + 20)
+                  setPokemonsDisplay(pokemonSlice)
+                }
+              }}
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js:122]: please define a handler function for this one instead of inline syntax

Comment (raymond-le-goldenowl) [reply src/pages/home-page.js:122]: i will fix that

File: src/pages/home-page.js [hunk 7]
```diff
@@ -46,41 +53,107 @@ export default function HomePage() {
...
+            />
+          </div>
+
+          <div className="col-2 d-flex align-items-end">
+            <select
+              style={{
+                minWidth: '100px',
+                height: '50px',
+                borderRadius: '50px'
+              }}
```
Comment (jun-nguyen-goldenowl) [inline src/pages/home-page.js:132]: don't use inline style

**Review by jun-nguyen-goldenowl — CHANGES_REQUESTED** [review]

**Review by jun-nguyen-goldenowl — APPROVED** [review]
please fix code review comments in PR #11

**Pull request merged**